### DISCUSSION TITLE
Valheim server crash on startup

### DISCUSSION THREAD

**Discussion by dementedusa** [GameServerManagers/LinuxGSM#3240] [General]
The server crashes almost instantly upon start.
console log:  https://pastebin.com/5hdDypEC
post details:  https://termbin.com/yfex
I have tried just starting it with stock configs.  my system is up to date.  this one is stumping me.

**Comment by SebPikPik** [comment]
Same here :(

**Comment by thegreatstorm** [comment]
What OS/distro you are using?

**Comment by thegreatstorm** [comment]
I attempted to replicate with a fresh install of ubuntu 20 and encountered no issues.

**Comment by dementedusa** [comment]
I am using unbuntu 18.04

**Comment by Novido** [comment]
Got the same issue on Ubuntu Server 20.04.2 LTS

EDIT: Tried 20.10 also with no success.

**Comment by SebPikPik** [comment]
I got the problem and it worked by giving a 5+ character password

**Comment by geoff10e** [comment]
This issue is caused by the 32bit version of steamclient.so being used in the install. Thankfully the 64bit version of steamclient.so is also included. I resolved this issue for my install by performing the following commands to put the correct version of the file in server files:

cp ~/valheim/serverfiles/steamclient.so ~/valheim/serverfiles/steamclient.bak
cp ~/valheim/serverfiles/linux64/steamclient.so ~/valheim/serverfiles/steamclient.so

**Reply by ghost** [reply]
worked here, thanks so much!!

**Comment by gbschenkel** [comment]
> This issue is caused by the 32bit version of steamclient.so being used in the install. Thankfully the 64bit version of steamclient.so is also included. I resolved this issue for my install by performing the following commands to put the correct version of the file in server files:
> 
> cp ~/valheim/serverfiles/steamclient.so ~/valheim/serverfiles/steamclient.bak
> cp ~/valheim/serverfiles/linux64/steamclient.so ~/valheim/serverfiles/steamclient.so

Didn't work for me.

**Comment by geoff10e** [comment]
@gbschenkel The error present in the OP, that's preventing the server from starting is the following segment:

> dlopen failed trying to load:
> steamclient.so
> with error:
> steamclient.so: wrong ELF class: ELFCLASS32
> [S_API FAIL] SteamAPI_Init(): Sys_LoadModule failed to load: steamclient.so
> [S_API FAIL] SteamAPI_Init() failed
> [Steamworks.NET] GameServer.Init() failed. 
> (Filename: ./Runtime/Export/Debug/Debug.bindings.h Line: 35)

My proposed fix will only work for users getting the same error as @dementedusa, the creator of this specific issue report.

**Comment by gbschenkel** [comment]
Well I found it needs at least use 2GB RAM, since I am running inside a container, the server was breaking while allocating memory. Now is up and running.

![image](https://user-images.githubusercontent.com/1715069/107442783-656c9700-6b16-11eb-9ff7-d40cd9f83de2.png)

**Comment by Novido** [comment]
> Well I found it needs at least use 2GB RAM, since I am running inside a container, the server was breaking while allocating memory. Now is up and running.
> 
> ![image](https://user-images.githubusercontent.com/1715069/107442783-656c9700-6b16-11eb-9ff7-d40cd9f83de2.png)

Yeah I bumped mine up to 4Gb RAM and 2 CPU for it to start.

**Comment by thegreatstorm** [comment]
Valheim 
* normal `2.3gb - 2.9gb`
* high pop(8-10) `3.2gb - 3.9gb`

**Comment by jonneymendoza** [comment]
hi, how did u get logs?

**Comment by geoff10e** [comment]
> 
> 
> hi, how did u get logs?

Logs should be located here:

~/valheim/log/console

**Comment by geoff10e** [comment]
@gbschenkel @Novido 
Can you confirm whether you were receiving the same error in the logs, as the reporter of this issue in the console, or were you getting another error?

**Comment by Azengar** [comment]
> I got the problem and it worked by giving a 5+ character password

You absolute mad lad! This is so stupid though...

**Comment by dementedusa** [comment]
> This issue is caused by the 32bit version of steamclient.so being used in the install. Thankfully the 64bit version of steamclient.so is also included. I resolved this issue for my install by performing the following commands to put the correct version of the file in server files:
> 
> cp ~/valheim/serverfiles/steamclient.so ~/valheim/serverfiles/steamclient.bak
> cp ~/valheim/serverfiles/linux64/steamclient.so ~/valheim/serverfiles/steamclient.so

@geoff10e that worked perfectly.  thank you very much.

**Reply by patmann03** [reply]
Does this have any impact on performance?  I'm seeing 20-25% CPU usage.  Server does seem to be running just fine using the base install

**Comment by wp1g19** [comment]
Same issue, but on windows, any tailored fix?

**Comment by gOOvER** [comment]
The server NEEDS a Password to start

**Comment by petahbred** [comment]
Also wanted to chime in, I had moved my saved world over from my local server, and the file was quite large, 3-4 players (only 2 now), it would not boot up until I increased the memory to 4GB

**Comment by C74K** [comment]
dont work bug exist server dont start
